src/io/config_auto.cpp

```diff
@@ -419,6 +421,12 @@ void Config::GetMembersFromString(const std::unordered_map<std::string, std::str
 
   GetString(params, "monotone_constraints_method", &monotone_constraints_method);
 
+  GetDouble(params, "monotone_penalty", &monotone_penalty);
+  CHECK_GE(monotone_penalty, 0.0);
+  if (max_depth > 0) {  //FIXME Not specified in config.h because I don't know how to specify that, please advise
```

Comment:
Please advise on this.

Reply:
Move thin into config.cpp, the check conflict function